taiga_halo2/src/constant.rs

```diff
@@ -67,29 +69,27 @@ pub const RESOURCE_LOGIC_CIRCUIT_PUBLIC_INPUT_NUM: usize =
     RESOURCE_LOGIC_CIRCUIT_MANDATORY_PUBLIC_INPUT_NUM
         + RESOURCE_LOGIC_CIRCUIT_CUSTOM_PUBLIC_INPUT_NUM
         + RESOURCE_LOGIC_CIRCUIT_RESOURCE_ENCRYPTION_PUBLIC_INPUT_NUM;
-pub const RESOURCE_LOGIC_CIRCUIT_MANDATORY_PUBLIC_INPUT_NUM: usize = 9;
+pub const RESOURCE_LOGIC_CIRCUIT_MANDATORY_PUBLIC_INPUT_NUM: usize = 6;
```

Comment:
Do we still want this RESOURCE_LOGIC_CIRCUIT_MANDATORY_PUBLIC_INPUT_NUM?

Reply:
Same for RESOURCE_LOGIC_CIRCUIT_CUSTOM_PUBLIC_INPUT_NUM and RESOURCE_LOGIC_CIRCUIT_RESOURCE_ENCRYPTION_PUBLIC_INPUT_NUM

Reply:
Are these now an upper bound instead of a mandatory strict number?

Reply:
yes, we still need the mandatory strict number.
Variable resource number wouldn't affect the number of resource logic public inputs. Because only the root(one element) of the resources go to the public inputs.